Blink/SceneDelegate.swift

```diff
@@ -454,3 +320,160 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
+        term.termDevice.write(sshCommand)
+      }
+    } else {
+      // No running command or shell found running, run the SSH command on the
+      // available shell
+      term.termDevice.write(sshCommand)
+    }
+  }
+  
+  /**
+   Handles the x-callback-url, if  a successful `x-success` URL is provided when being called from apps like Shortcuts it returns to the original app after a successful execution.
+    - Parameters:
+      - xCallbackUrl: The x-callback-url specified by the user, URL format should be `blinkshell://run?key=KEY&cmd=CMD%20ENCODED`
+   */
+  func handleXcallbackUrl(with xCallbackUrl: URL) {
+    
+    if let xCallbackUrlHost = xCallbackUrl.host, xCallbackUrlHost == "run" {
+      
+      let components = URLComponents(url: xCallbackUrl, resolvingAgainstBaseURL: true)
+      
+      var xCancelURL: URL?
+      var xSuccessURL: URL?
+      var xErrorURL: URL?
+      
+      guard let items = components?.queryItems else {
+        return
+      }
+      
+      if let xCancel = items.first(where: { $0.name == "x-cancel" })?.value {
+        xCancelURL = URL(string: xCancel)
+      }
+
+      if let xError = items.first(where: { $0.name == "x-error" })?.value {
+        xErrorURL = URL(string: xError)
+      }
+      
+      if let xSuccess = items.first(where: { $0.name == "x-success" })?.value {
+        xSuccessURL = URL(string: xSuccess)
+      }
+      
+      if !BKDefaults.isXCallBackURLEnabled() {
+        if let xCancelURL = xCancelURL {
+          blink_openurl(xCancelURL)
+        }
+        return
+      }
+      
+      // Cancel execution of the command if the x-callback-url doesn't have a
+      // key field present that is needed to allow URL actions
+      guard let keyItem: String = items.first(where: { $0.name == "key" })?.value else {
+        if let xCancelURL = xCancelURL {
+          blink_openurl(xCancelURL)
+        }
+        
+        return
+      }
+      
+      // Cancel the execution of the command as x-callback-url are not
+      // enabled for the user's or the x-callback-url does not have
+      // the correct key set
+      if keyItem != BKDefaults.xCallBackURLKey() {
+        if let xCancelURL = xCancelURL {
+          blink_openurl(xCancelURL)
+        }
+        
+        return
+      }
+      
+      guard let cmdItem: String = items.first(where: { $0.name == "cmd" })?.value else {
+        if let xErrorURL = xErrorURL {
+          blink_openurl(xErrorURL)
+        }
+        return
+      }
+      
+      let spCtrl = _spCtrl
+      
+      guard let term = spCtrl.currentTerm() else {
+        if let xErrorURL = xErrorURL {
+          blink_openurl(xErrorURL)
+        }
+        return
+      }
+      
+      spCtrl.focusOnShellAction()
+      
+      // If SSH/mosh connection is already open in the current terminal shell
+      // create a new one and then write the SSH command
+      if term.isRunningCmd() {
```

Comment:
also better guard here and return after `term.xCallbackLineSubmitted(cmdItem, xSuccessURL)`

so we quit method early